derive/src/parse.rs

```diff
@@ -444,6 +452,99 @@ fn next_enum(mut source: &mut Peekable<impl Iterator<Item = TokenTree>>) -> Enum
     }
 }
 
+fn get_bounds(source: &mut Peekable<impl Iterator<Item = TokenTree>>) -> HashSet<String> {
+    let mut ret = HashSet::new();
+    let mut tmp: Vec<String> = Vec::new();
+
+    match source.peek() {
+        Some(content) if content.to_string() == "<" => {
+            source.next();
+            let mut typename = "".to_string(); 
+            let mut in_type = true;
+            while let Some(tok) = source.next() {
+                tmp.push(tok.to_string());
+                match tok.to_string().as_str() {
+                    ">" => {
+                        if !typename.is_empty(){
+                            ret.insert(typename.clone());
+                        }
+                        break
+                    },
+                    ":" => {
+                        ret.insert(typename.clone());
+                        typename.clear();
+                        in_type = false
+                    },
+                    "," => {
+                        if in_type {
+                            ret.insert(typename.clone());
+                            typename.clear();
+                        }
+                        in_type = true
+                    },
+                    c => {
+                        if in_type {
+                            typename+=c
+                        } 
+                    }
+                }
+            }
+        },
+        _ => ()
+    }
+    ret
+}
+
+pub(crate) fn struct_bounds_strings(struct_: &Struct, bound_name: &str) -> (String, String) {
+    let mut generics: Vec<Type> = Vec::new();
+    for field in struct_.fields.iter() {
+        if field.ty.is_generic {
+            generics.push(field.ty.clone())
+        }
+    }
+    if generics.is_empty() {
+        return ("".to_string(), "".to_string());
+    }
+    let mut generic_w_bounds = "<".to_string();
+    for generic in generics.iter() {
+        generic_w_bounds += &format!("{}: nanoserde::{}, ", generic.path, bound_name);
+    }
+    generic_w_bounds += ">";
+
+    let mut generic_no_bounds = "<".to_string();
+    for generic in generics.iter() {
+        generic_no_bounds += &format!("{}, ", generic.path);
+    }
+    generic_no_bounds += ">";
+    return (generic_w_bounds, generic_no_bounds);
+}
+
+pub(crate) fn enum_bounds_strings(enum_: &Enum, bound_name: &str) -> (String, String) {
```

Comment:
I feel like with "generics" field within `Enum`/`Struct` structs we might end up without an "is_generic" field in the "Type" and this function will not be required. Sort of like in "syn": https://docs.rs/syn/0.12.15/syn/struct.ItemStruct.html, the  "generis" field. 

I am not really sure about this :)

Reply:
Alright, understand if it's not what you wanted. I'm not sure I'm seeing how that would avoid the linked function though. The actual impl block would still need the bounds I thought

Reply:
I would start with adding "generics" field to the "Struct" struct and checking if this would be enough to implement "derive_de_bin_struct". Than I would check how hard it its to get the "generics" field data while parsing the struct

Reply:
Sounds good, will give this a shot when I have time again

Reply:
That should be done now

Reply:
@not-fl3 Any further issues with this pr? The bounds function exits early for non generic structs, so shouldn't really add much to compile times